packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/bitmap.dart

```diff
@@ -44,14 +51,14 @@ const double _naturalPixelRatio = 1.0;
 /// a default marker icon.
 /// Use the [BitmapDescriptor.defaultMarkerWithHue] to create a
 /// [BitmapDescriptor] for a default marker icon with a hue value.
-class BitmapDescriptor {
-  const BitmapDescriptor._(this._json);
+abstract class BitmapDescriptor {
+  const BitmapDescriptor._();
 
   /// The inverse of .toJson.
   // TODO(stuartmorgan): Remove this in the next breaking change.
   @Deprecated('No longer supported')
-  BitmapDescriptor.fromJson(Object json) : _json = json {
-    assert(_json is List<dynamic>);
+  static BitmapDescriptor fromJson(Object json) {
```

Comment:
Why not use the `: _json` syntax?

Reply:
This PR removes the `_json` field from `BitmapDescriptor`, and since it also makes `BitmapDescriptor` an abstract class with a concrete subtype for each possible set of member data, `fromJson` needs to return an instance of one of those subtypes.